js/interface/add-markers.js

```diff
@@ -86,40 +87,55 @@ Fliplet.InteractiveMap.component('add-markers', {
       return this.widgetData.maps[this.selectedMapIndex]
     }
   },
-  watch: {
-    markersDataSource(ds, oldDs) {
-      // If the user is changes away from the data source the component creates
-      if (!ds || !oldDs || (ds.id !== this.dataSourceId && !this.dataWasChanged)) {
+  methods: {
+    markersDataSourceChange(ds) {
+      if (!ds || !this.markersDataSource || (ds.id !== this.dataSourceId && !this.dataWasChanged)) {
         return Fliplet.Modal.confirm({
           title: 'Changing data source',
           message: '<p>If you change the data source, the one we created for you will be deleted.<br>Are you sure you want to continue?</p>'
         }).then((result) => {
           if (!result) {
-            this.markersDataSource = oldDs
-            this.dataSourceId = oldDs.id
             return
           }
 
-          // Keep ref to delete ds later on save
-          this.dataSourceToDelete = oldDs.id
-
-          this.dataSourceId = ds.id
-          this.dataWasChanged = true
+          this.dataSourceToDelete = this.dataSourceId
 
-          // Remove from dataSources
           this.dataSources = _.filter(this.dataSources, (dataSource) => {
-            return dataSource.id !== oldDs.id
+            return dataSource.id !== this.dataSourceId
           })
-
-          // Resets select fields
           this.resetSelectFields()
+
+          this.dataSourceId = ds.id
+          this.markersDataSource.columns = ds.columns
+
+          this.dataWasChanged = true
         })
       }
+    },
+    initDataSourceProvider(currentDataSourceId) {
+      const $vm = this
+      let dataSourceData = {
+        dataSourceTitle: 'Markers data source',
+        dataSourceId: currentDataSourceId,
+        appId: Fliplet.Env.get('appId'),
+        default: {
+          name: 'Profile data for ' + Fliplet.Env.get('appName'),
+          entries: [],
+          columns: []
+        },
+        accessRules: []
+      }
 
-      this.dataSourceId = ds.id
-    }
-  },
-  methods: {
+      this.dataSourceProvider = Fliplet.Widget.open('com.fliplet.data-source-provider', {
+        selector: '#dataSourceProvider',
+        data: dataSourceData,
+        onEvent: function(event, dataSource) {
```

Comment:
Use ES6 arrow functions so the context doesn't change inside the function.

```suggestion
        onEvent: (event, dataSource) => {
```